tests.py

```diff
@@ -24,12 +40,29 @@ def run_vsm(self, name, input_data, expected_output, use_initial=True):
             cmd += ['--initial-state={}'.format(initial_state)]
 
         cmd += [conf]
+        if TestVSM.ipc_module:
+            cmd += [ '--ipc-module={}'.format(TestVSM.ipc_module) ]
 
-        process = Popen(cmd, stdin=PIPE, stdout=PIPE)
+        if TestVSM.ipc_module:
+            process = Popen(cmd)
 
-        output, _ = process.communicate(input=input_data.encode(), timeout=2)
+            for signal, value in format_ipc_input(input_data):
+                send(signal, value)
 
-        self.assertEqual(output.decode() , expected_output)
+            # Workaround for signals expecting ''
+            if expected_output == '':
```

Comment:
I think you can send quit unconditionally? That won't be a problem for the other cases, right?